tests/integration-tests/framework/fixture_utils.py

```diff
@@ -122,6 +124,18 @@ def release(self):
                         self._save_fixture_data(data)
                 data = self._load_fixture_data()
 
+            if time.time() > timeout:
+                logging.error(
+                    (
+                        "Shared fixture %s has not been released in 4 hours, "
+                        "currently in use by %d processes (%s), destroying it."
+                    ),
+                    self.name,
+                    data.counter,
+                    data.currently_using_processes,
+                )
+                break
```

Comment:
thought (non-blocking): what about to kill the "hanging" pid?

Reply:
good call-out, addressed.